src/Transactions/AccountSignatureMap.cs

```diff
@@ -56,4 +56,110 @@ public Grpc.V2.AccountSignatureMap ToProto()
         }
         return accountSignatureMap;
     }
+
+    internal static AccountSignatureMap From(Grpc.V2.AccountSignatureMap map)
+    {
+        var dict = new Dictionary<AccountKeyIndex, byte[]>();
+        foreach (var s in map.Signatures)
+        {
+            dict.Add(new AccountKeyIndex((byte)s.Key), s.Value.Value.ToByteArray());
+        }
+        return Create(dict);
+    }
+
+    /// <summary>Check for equality.</summary>
+    public bool Equals(AccountSignatureMap? other) => other != null &&
+               other.GetType().Equals(this.GetType()) &&
+               this.Signatures.Count == other.Signatures.Count &&
+               this.Signatures.All(p => p.Value == other.Signatures[p.Key]);
+
+    /// <summary>Gets hash code.</summary>
+    public override int GetHashCode()
+    {
+        // Based on https://stackoverflow.com/questions/1646807/quick-and-simple-hash-code-combinations
+        unchecked
+        {
+            var hash = 17;
+            foreach (var (key, val) in this.Signatures)
+            {
+                hash = (hash * 31) + key.GetHashCode();
+                hash = (hash * 31) + Helpers.HashCode.GetHashCodeByteArray(val);
+            }
```

Comment:
**Issue:** `Signatures` is an unordered collection, and I am not sure if we can rely on the ordering being the same, for two equal signatures.
We need this hash code to be independent of the iteration ordering.